### ISSUE TITLE
[Version 1.5.2] 'obj' doesn't override 'objective' for XGboost

### ISSUE THREAD

**Issue by ganzheng85** [dmlc/xgboost#8523]
I did experiment to test 'obj'. I put log loss function for 'obj', and in the 'objective' parameter, objective = 'reg:squarederror' and objective = 'binary:logistic' give me different results. 

Since I already assigned 'obj', 'objective' should be overwritten and wouldn't impact the results, alright?

Why 'objective' parameter affects 'obj'? Thanks.

**Comment by trivialfis** [comment]
Hi, thank you for raising the issue. In short, this is a known limitation.

Longer explanation: XGBoost does transformation based on objective, when a custom objective is supplied, the default internal objective written in c++ rmse has no transformation and hence doesn't stay in the way of the custom objective function defined in Python/R/Scala.

As a result, don't specify the objective parameter if a custom one is provided.
